changelog/59026.fixed

```diff
@@ -0,0 +1 @@
+Add extra onfail req check in the state engine to allow onfail to be used with onchanges and other reqs in the same state
```

Comment:
If this is changing behavior we should also name the changelog file 59026.changed

Reply:
Agreed